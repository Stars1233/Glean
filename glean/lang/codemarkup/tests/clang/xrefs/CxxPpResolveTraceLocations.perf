{
  "@generated": null,
  "facts_searched": { },
  "full_scans": [
    "codemarkup.cxx.CxxResolveTraceLocation.4",
    "codemarkup.pp.PpResolveTraceLocation.3"
  ]
}