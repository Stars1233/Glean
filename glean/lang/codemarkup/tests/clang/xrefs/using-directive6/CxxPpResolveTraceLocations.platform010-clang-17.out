[
  "@generated",
  {
    "key": {
      "tuplefield0": {
        "name": "N1",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 9,
            "columnBegin": 1,
            "lineEnd": 13,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 231, "length": 2 }
        }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "N1" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 13,
                      "columnEnd": 1
                    }
                  }
                },
                "members": {
                  "key": [
                    {
                      "record_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "T" },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "N1" } }
                                }
                              }
                            }
                          },
                          "kind": { "struct_": { } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 11,
                            "columnBegin": 1,
                            "lineEnd": 11,
                            "columnEnd": 11
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "N1",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 15,
            "columnBegin": 1,
            "lineEnd": 15,
            "columnEnd": 18
          }
        }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "usingDirective": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "N1" },
                    "scope": { "global_": { } }
                  }
                },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 15,
                  "columnBegin": 1,
                  "lineEnd": 15,
                  "columnEnd": 18
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "N1",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 17,
            "columnBegin": 1,
            "lineEnd": 23,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 285, "length": 2 }
        }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "N1" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 17,
                      "columnBegin": 1,
                      "lineEnd": 23,
                      "columnEnd": 1
                    }
                  }
                },
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "foo" } } },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "N1" } }
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [ ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 19,
                            "columnBegin": 1,
                            "lineEnd": 21,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "T",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 11,
            "columnBegin": 1,
            "lineEnd": 11,
            "columnEnd": 11
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 244, "length": 1 }
        }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "T" },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "N1" } } }
                        }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 11,
                      "columnBegin": 1,
                      "lineEnd": 11,
                      "columnEnd": 11
                    }
                  }
                },
                "bases": [ ],
                "members": { "key": [ ] }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "foo",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 19,
            "columnBegin": 1,
            "lineEnd": 21,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 296, "length": 3 }
        }
      },
      "tuplefield1": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "foo" } } },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "N1" } } }
                        }
                      }
                    },
                    "signature": {
                      "key": { "returns": { "key": "void" }, "parameters": [ ] }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 19,
                      "columnBegin": 1,
                      "lineEnd": 21,
                      "columnEnd": 1
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "tuplefield0": {
        "name": "x",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 20,
            "columnBegin": 3,
            "lineEnd": 20,
            "columnEnd": 7
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 310, "length": 1 }
        }
      },
      "tuplefield1": {
        "cxx": {
          "decl": {
            "variable": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "x" },
                    "scope": {
                      "local": {
                        "key": {
                          "name": { "key": { "name": { "key": "foo" } } },
                          "scope": {
                            "namespace_": { "key": { "name": { "key": "N1" } } }
                          }
                        }
                      }
                    }
                  }
                },
                "type": { "key": "N1::T" },
                "kind": { "local": { "kind": 0, "attribute": 0 } },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 20,
                  "columnBegin": 3,
                  "lineEnd": 20,
                  "columnEnd": 7
                }
              }
            }
          }
        }
      }
    }
  }
]